File: 3k/1s/cpp/conversation/Realist.cpp
class Realist: public Person
{
protected:
    char* getTypeText()
    {
        return "Реалист";
    }

public:
    Realist(char* newName, int newAge)
    {
        init(newName, newAge);
    }

    const void Hello(Person* &personTwo) {
        if (abs(personTwo->getAge() - getAge()) <= 5) {
            cout << "Привет, " << personTwo->getName();
        } else {
            cout << "Здравствуйте, " << personTwo->getName();
        }
    }
};

File: 3k/2s/java/5. Conversation/src/com/company/Realist.java
package com.company;

public class Realist extends Person
{
    Realist(String newName, int newAge, int newSex)
    {
        super(newName, newAge, newSex);
    }

    public String getTypeText()
    {
        return "Реалист" + getTypeTextSuffix();
    }

    public void hello(Person twoPerson) {
        if (Math.abs(twoPerson.getAge() - getAge()) <= 5) {
            System.out.print("Привет, " + twoPerson.getName());
        } else {
            System.out.print("Здравствуй, " + twoPerson.getName());
        }
    }
}
